plugins/sonarqube-issues/README.md

```diff
@@ -27,22 +27,41 @@ info:
 This plugin requires a proxy to SonarQube. To set up:
 
 - Create a token in SonarQube by clicking on your profile > My Account > Security
-- In Cortex, define a secret whose value is your new token. Name it `sonarqube_plugin`.
+- In Cortex, define a secret whose value is your new token. Name it `sonarqube_secret`.
 - Create a proxy:
 
   - Navigate to Plugins, then click on the Proxies tab, then click on Create Proxy
   - Give the proxy a name, then click on Add URL
-  - For the URL Prefix, type in the base URL of your SonarQube instance. Default for cloud is `https://sonarcloud.io`. **This URL prefix should be exactly the same as the value of the baseURL variable in [SonarqubeIssues.tsx](src/components/SonarqubeIssues.tsx) - If you are self-hosting SonarQube, you will have to put your own base URL in both places!**
-  - Click on Add Header and add a header whose name is `Authorization` and whose value is `Bearer {{secrets.sonarqube_plugin}}` (include the curly braces!)
+  - For the URL Prefix, type in the API base URL of your SonarQube instance. Default for cloud is `https://sonarcloud.io`. If you are self-hosting SonarQube, you will have a different API base URL.
+  - Click on Add Header and add a header whose name is `Authorization` and whose value is `Bearer {{secrets.sonarqube_secret}}` (include the curly braces!)
 
 - Once you are done, the proxy should look like the below:
 
 <div align="center"><img src="img/sonarqube-proxy.png"></div>
 
+### Self-Hosted setup
+
+The plugin uses `https://sonarcloud.io` as its default API base URL. If you are self-hosting Sonarqube, then you will have a different URL. To configure the plugin to use that URL, you can create a Sonarqube plugin configuration entity in Cortex with your own API base URL.
+
+- Consider creating a new entity type, so that any existing scorecards are not affected by ths configuration entity. In this example, we have created a new entity type called `plugin-configuration`
+- Create a new entity with the tag `sonarqube-plugin-config`
+- Set `x-cortex-definition.sonarqube-api-url` to the value of your ServiceNow Instance URL. For example, if my Sonarqube API base URL was `https://sonarqube.martindstone.com`, my `sonarqube-plugin-config` entity would look like this:
+
+```yaml
+openapi: 3.0.1
+info:
+  title: Sonarqube Plugin Config
+  description: ""
+  x-cortex-tag: sonarqube-plugin-config
+  x-cortex-type: plugin-configuration
+  x-cortex-definition:
+    sonarqube-api-url: https://sonarqube.martindstone.com
```

Comment:
Docs ask for the sonarqube-api-url key, but the code is checking for sonarqube-key:

<img width="646" alt="image" src="https://github.com/user-attachments/assets/df57197e-8d19-4d52-b1ff-fd9201359011">

Reply:
One more tweak and then I'm good to sign off! At the beginning, the README specifies:  

`If the cortex.yaml has a SonarQube Project defined in its x-cortex-static-analysis configuration, it will query for issues pertaining to that project. For example:`

And the example shows:

`openapi: 3.0.1
info:
  title: Funrepo
  description: it is a fun repo
  x-cortex-git:
    github:
      alias: cortex
      repository: martindstone-org/funrepo
  x-cortex-tag: funrepo
  x-cortex-type: service
  x-cortex-static-analysis:
    sonarqube:
      **project: martindstone-org_funrepo`**
      
  I think we need to call out specifically that the value in x-cortex-static-analysis.sonarqube.project needs to specifically match the *project key* in SonarQube and not the project name. That tripped me up once I got the authentication working, I started using the project name and it just wasn't retrieving issues, switched to project key and it works like a charm!!

<img width="640" alt="image" src="https://github.com/user-attachments/assets/a15a076d-77db-41ab-9c77-c4cbaa0569f2">

<img width="1204" alt="image" src="https://github.com/user-attachments/assets/571fc123-a1d2-44b2-90e1-38aa0667badc">